### PR TITLE
Mongo db Configuration

### PR DESCRIPTION
**Pull request by filg68**
This commit creates a configuration file that uses environment variables to define the location of your mongo DB as well as the names of the dev and test databases.  I went with evergreen_dev and evergreen_test but you can override the defaults by setting your own .env variables.

**Pre-Requisites:**
[Install MongoDB](https://docs.mongodb.com/manual/installation/) on your machine by following the instructions for your os type.  My installation was for version 4.2.

**To test on your machine:**
1. Go to app\server\ directory
2.  Download the new code - `git pull origin MongoDB`
3. Run ` npm install` to updated required packages (mongoose and dotenv)
4. From terminal start mongoDB - (script for Mac can be run by running `npm run start-mongo-db-on-mac` - if you've installed it in the standard directory, otherwise follow instructions for your os\build).  A successful launch of MongoDB will leave you with a hung terminal window with no status message - if you see errors fix them ... :-). 
5. In a new terminal windows go to app\server directory.
6. Run `node run test`.  This will run the unit tests for the model file and the db-connect file.  All 5 tests should pass.

Some additional tools, I downloaded compass which is a gui for using MongoDB.  You can find download instructions [here ](https://www.mongodb.com/products/compass).  Download the community version.

**Additional Notes**
1.  The default install of MongoDB does not require any access permissions or authorization - I find this disturbing but didn't want to mess with it - but I figure at some point we will want to have some sort of authorization to the database.
2. I look forward to the day where writing 10 lines of code does not require 6 hours of research :-).

### CODE REVIEW

**Comment by filg68** [comment]
#10 - Resolves issue

File: server/config/db-connect.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//db-connect.js
+/* Created by Fil on August 25, 2019 */
+
+//TODO (FG) - Change require to import once Babel works?
+const config = require('./config.js');
+const mongoose = require('mongoose');
+
+const dbUrl = `${config.db.dbType}://${config.db.host}:${config.db.port}/${config.db.name}`;
+console.log (dbUrl);
```
Comment (skassam21) [inline server/config/db-connect.js]: probably want to remove this console.log

File: server/config/db-connect.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+//db-connect.js
+/* Created by Fil on August 25, 2019 */
+
+//TODO (FG) - Change require to import once Babel works?
```
Comment (skassam21) [inline server/config/db-connect.js]: does babel not work if you run npm run dev?

File: server/config/db-connect.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  return mongoose.connect(dbUrl)
+};
+
+const user = new mongoose.Schema({
+  firstName: String,
+  lastName:String,
+  eMail:String
+});
+
+const User = mongoose.model('user', user);
```
Comment (skassam21) [inline server/config/db-connect.js]: This should probably be in it's own file for a user model

File: server/config/db-connect.js (hunk 1 above)
Comment (filg68) [reply server/config/db-connect.js]: oops ... coming out now.

File: server/config/db-connect.js (hunk 2 above)
Comment (filg68) [reply server/config/db-connect.js]: No I am getting an error on running the server.  My assumption was that it was Babel but haven't really dug in to it. The client starts up fine.

File: server/config/db-connect.js (hunk 3 above)
Comment (filg68) [reply server/config/db-connect.js]: Ok, I will create a models directory and move it in there.

Comment (filg68) [reply server/config/db-connect.js]: I refactored the model into its own file and added unit tests for the model and db-connect files.  You can run tests by running `npm run test`

**Comment by filg68** [comment]
You can now run unit tests of the User.js model and the db-connect.js mongoose connector by running `npm run test`

**Pull request merged**